wordpress-mobile/WordPress-iOS
# Notifications Control Center: Disabled Push Notifications

#### Notification Settings: Disabled Push Notifications
The goal of this PR is to implement support to properly handle scenarios in which Push Notifications are disabled, so that the new `Notifications Control Center` interface reacts accordingly.

Below you may find a detailed list of the test scenarios required to validate this functionality.

--

#### Test Scenario A: Disabled Push Notifications in iOS 7 
1. Launch WPiOS in a iOS 7 device (OR simulator)
2. Make sure that Push Notifications for WPiOS are disabled
3. Open `Me` > `Notifications`
4. Attempt to open the `Push Notification` Settings for any of your blogs

As a result, you should get a nice AlertView indicating that Push Notifications are disabled. Switch to Settings.app and enable Push Notifications.

After going back to WPiOS, the Push Notifications row should be enabled.


--

#### Test Scenario B: Disabled Push Notifications in iOS 8
1. Launch WPiOS in a iOS B device (OR simulator)
2. Make sure that Push Notifications for WPiOS are disabled
3. Open `Me` > `Notifications`
4. Open the `Push Notification` settings of any of your blogs

As a result, the interface should display a row that reads `Go to iPhone Settings`, and the tableView footer should detail what's going on..

Please, go to Settings.app, enable Push Notifications, and bring WPiOS to foreground. Note that the view should refresh automatically, and you should be able to update the settings.

--

#### Test Scenario C: Enabled Push Notifications
1. Launch WPiOS in a iOS B device (OR simulator)
2. Make sure that Push Notifications for WPiOS are **enabled**
3. Open `Me` > `Notifications`
4. Open the `Push Notification` settings of any of your blogs

Note that the app should allow you to update anything, and no popup (or tableView Footer) should show up. Please, verify in both iOS 7 and 8.


--


Reference: Issue #3807
